repos/system_upgrade/el7toel8/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py

```diff
@@ -60,5 +65,36 @@ def get_booted_kernel():
         )
 
 
-def get_target_version():
-    return os.getenv('LEAPP_DEVEL_TARGET_RELEASE', None) or CURRENT_TARGET_VERSION
+def get_target_version(flavour=LEAPP_UPGRADE_FLAVOUR_DEFAULT):
+    current_target_version = CURRENT_TARGET_VERSION
+    if flavour == LEAPP_UPGRADE_FLAVOUR_SAP_HANA:
+        current_target_version = CURRENT_SAP_HANA_TARGET_VERSION
+    return os.getenv('LEAPP_DEVEL_TARGET_RELEASE', None) or current_target_version
+
+
+HANA_BASE_PATH = '/hana/shared'
+HANA_SAPCONTROL_PATH = 'exe/linuxx86_64/hdb/sapcontrol'
+HANA_SAPHOSTCTRL_PATH = '/usr/sap/hostctrl/exe/saphostctrl'
+
+
+def detect_hana_shared_sapcontrol():
+    """ Detect SAP Hana based on existance of /hana/shared/*/exe/linuxx86_64/hdb/sapcontrol """
+    if os.path.exists(HANA_BASE_PATH):
+        for entry in os.listdir(HANA_BASE_PATH):
+            # Does /hana/shared/{entry}/exe/linuxx86_64/hdb/sapcontrol exist?
+            if os.path.exists(os.path.join(HANA_BASE_PATH, entry, HANA_SAPCONTROL_PATH)):
+                return True
+    return False
+
+
+def detect_sap_hana():
+    """ Detect SAP Hana either by existance of saphostctrl or sapcontrol """
+    if os.path.exists(HANA_SAPHOSTCTRL_PATH):
```

Comment:
if this detects all SAP stuff this and the line after need to go away.

So basically it would consist only of the call to detect_hana_shared_sapcontrol, that would detect SAP Hana only.

Reply:
you are right. let's do that.

Reply:
This part has been removed to ensure we only detect SAP Hana